pkg/controller/operands/cdi.go

```diff
@@ -134,13 +134,23 @@ func NewCDI(hc *hcov1beta1.HyperConverged, opts ...string) *cdiv1beta1.CDI {
 		hc.Spec.Workloads.NodePlacement.DeepCopyInto(&spec.Workloads)
 	}
 
+	if err := applyPatchToSpec(hc, common.JSONPatchCDIAnnotationName, &spec); err != nil {
+		return nil, err
+	}
+
+	cdi := NewCDIWithNameOnly(hc, opts...)
+	cdi.Spec = spec
+
+	return cdi, nil
+}
+
+func NewCDIWithNameOnly(hc *hcov1beta1.HyperConverged, opts ...string) *cdiv1beta1.CDI {
```

Comment:
`NewCDI` now returns error now. The `NewCDIWithNameOnly` method doesn't return error, and creates an empty CDI, except for name. It may be used to retriving CDI from K8s. For now it used only within `NewCDI()` but it can later replace several places that now calls the original GetCDI, causing all the logic to run, when the only usage is to get the name and the namespace. The same was done for KV and CNAO.